GDJS/GDJS/IDE/ExporterHelper.cpp

```diff
@@ -852,6 +853,168 @@ const std::array<int, 20> IOS_ICONS_SIZES = {
 };
 const std::array<int, 6> ANDROID_ICONS_SIZES = {36, 48, 72, 96, 144, 192};
 
+const gd::String ExporterHelper::GenerateServiceWorker(
+    const std::vector<gd::String> includesFiles,
+    const std::vector<gd::String> resourcesNames) {
+  gd::String generatedIncludes = "[";
+  gd::String staticIncludes = "['/','/index.html',";
+  gd::String resources = "[";
+
+  for (const auto &includeName : includesFiles) {
+    if (fs.IsAbsolute(includeName)) {
+      // By convention, absolute files are
+      //   1. Not files bundled with GDevelop, but autogenerated code files that
+      //   may change between exports
+      //   2. Exported under their file name, ignoring the directory they
+      //   originate from.
+      generatedIncludes += "\"" + fs.FileNameFrom(includeName) + "\",";
+    } else {
+      // By convention, an include file that is relative is relative to
+      // the "<GDJS Root>/Runtime" folder, and therefore is not expected to
+      // change unless the GDevelop version is changed.
+      staticIncludes += "\"" + includeName + "\",";
+    }
+  };
+
+  for (const auto &resourceFile : resourcesNames)
+    resources += "\"" + resourceFile + "\",";
+
+  // Replace last "," with "]"
+  generatedIncludes.pop_back();
+  generatedIncludes += ']';
+  staticIncludes.pop_back();
+  staticIncludes += ']';
+  resources.pop_back();
+  resources += ']';
+
+  return gd::String(R"js_code(
+// While only one cache is present for now, this system is a best practice 
+// that would allow in the future to have a different cache invalidated at different times
+// e.g. GDJS Runtime code cache could be kept as long as the GDevelop version has not changed.
+const CURRENT_CACHES = [
+  {
+    name: 'gdjs-generated-code-vGENERATED_CODE_CACHE_VERSION',
+    files: /*GENERATED_CODE_FILES*/,
+  },
+  {
+    name: 'gdjs-static-code-vSTATIC_CODE_CACHE_VERSION',
+    files: /*STATIC_CODE_FILES*/,
+  },
+  {
+    name: 'gdjs-resources-vRESOURCES_CACHE_VERSION',
+    files: /*RESOURCE_FILES*/,
```

Comment:
It would be cool if sometime in the future we can do it on a file-pre-file basis and only invalidate the cache of those that have actually changed, but let's keep it simple for now.